express-notes-api/server.js

```diff
@@ -0,0 +1,95 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+const noteObj = await readFile('data.json', 'utf-8');
+const parsedObj = JSON.parse(noteObj);
+const notesObj = parsedObj.notes;
+const entryId = parsedObj.nextId;
+
+async function write(request) {
```

Comment:
This function looks like it should be reusable, but it's not. Lines 13 and 14 should only happen during create, not update or delete.

Because the function name looks reusable, you are likely to accidentally call it during future code maintenance when you shouldn't. I would suggest either renaming it so it doesn't look like it can be reused, or removing the non-reusable logic to make this reusable. The latter would have the benefit of letting you write line 15 only once.